hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/ReplaceCommitValidateUtil.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.table.action.cluster;
+
+import org.apache.hudi.common.model.HoodieReplaceCommitMetadata;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.util.JsonUtils;
+import org.apache.hudi.exception.HoodieException;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.apache.hudi.common.util.FileIOUtils.LOG;
+
+public class ReplaceCommitValidateUtil {
+  public static final String REPLACE_COMMIT_FILE_IDS = "replaceCommitFileIds";
+  public static void validateReplaceCommit(HoodieTableMetaClient metaClient) {
+    metaClient.reloadActiveTimeline();
+    Set<String> replaceFileids = new HashSet<>();
+
+    // Verify pending and completed replace commit
+    Stream.concat(metaClient.getActiveTimeline().getCompletedReplaceTimeline().getInstantsAsStream(),
+        metaClient.getActiveTimeline().filterInflights().filterPendingReplaceTimeline().getInstantsAsStream()).map(instant -> {
+          try {
+            HoodieReplaceCommitMetadata replaceCommitMetadata =
+                HoodieReplaceCommitMetadata.fromBytes(metaClient.getActiveTimeline().getInstantDetails(instant).get(),
+                HoodieReplaceCommitMetadata.class);
+            if (!instant.isCompleted()) {
+              return JsonUtils.getObjectMapper().readValue(replaceCommitMetadata.getExtraMetadata().get(REPLACE_COMMIT_FILE_IDS), String[].class);
+            } else {
+              return replaceCommitMetadata.getPartitionToReplaceFileIds().values().stream()
+                  .flatMap(List::stream)
+                  .toArray(String[]::new);
+            }
+          } catch (Exception e) {
+            // If the key does not exist or there is a JSON parsing error, LOG reports an error and ignores it.
+            LOG.error("Error when reading REPLACECOMMIT meta", e);
+            return null;
```

Comment:
In what cases are we not able to read the REPLACECOMMIT meta here? If we are not able to read the meta here, is it safe to proceed with the REPLACECOMMIT that the current writer is performing?

Reply:
The issue currently observed is that the key "REPLACE_COMMIT_FILE_IDS" does not have a value in the inflight metadata. In such cases, the code throws an error, but in certain testing scenarios, it may not call the generated inflight REPLACECOMMIT through `table.cluster.` Instead, it directly calls transitionReplaceRequestedToInflight to generate the inflight operation. Consequently, an exception occurs because REPLACE_COMMIT_FILE_IDS is not stored in the inflight metadata by `table.cluster`. No other issues have been identified thus far. The reason for catching this exception without terminating the program is that it serves as an additional validation program. I do not want to impact the original execution logic unless there is a validation failure. Therefore, any other unexpected exceptions are also skipped in this scenario. We can discuss how this situation can be handled better.  ^_^

Reply:
In other words, my expectation is that the program should not encounter any issues leading to termination in my validation, unless there are duplicate file ids.